server/base/src/main/java/org/apache/accumulo/server/init/Initialize.java

```diff
@@ -257,10 +256,20 @@ private static boolean createDirs(VolumeManager fs, InstanceId instanceId, Set<S
         fs.mkdirs(
             new Path(new Path(baseDir, Constants.VERSION_DIR), "" + AccumuloDataVersion.get()),
             new FsPermission("700"));
+        log.info("Created directory {}", baseDir);
+
         Path iidLocation = new Path(baseDir, Constants.INSTANCE_ID_DIR);
         fs.mkdirs(iidLocation);
-        fs.createNewFile(new Path(iidLocation, instanceId.canonical()));
-        log.info("Created directory {}", baseDir);
+        log.info("Created directory {}", iidLocation);
+
+        Path iidPath = new Path(iidLocation, instanceId.canonical());
+        boolean createSuccess = fs.createNewFile(iidPath);
+        if (!createSuccess || fs.exists(iidPath)) {
```

Comment:
This could be misleading if the file already exists. It looks like the only way it will fail without error is if the file already exists. So I think you could just do:
```suggestion
        if (fs.createNewFile(iidPath)) {
```

Reply:
Do we want to throw an error if the iidPath already exists? It could be verified somewhere else in `Initialize`

Reply:
I'm not sure want happens if it exists - didn't think about that.  The check was based on the following assumptions

  - createNewFile could return false on any hdfs error (forgot about exists) so we should check.
  - the `fs.exists()` was specifically added to read that yea, verily the file was really written as hdfs said.

If I could have found a sync method, I would have added that too.  Not really worried about performance of something that runs once per instance - and really trying to nail down if we have a timing issue or a concurrent test issue with the mini tests that claim they can find the instance.